tools/generate-release-details.sh

```diff
@@ -93,6 +93,21 @@ v_echo_ts() {
         [ -z "$DEBUG_X" ] || set $DEBUG_X
 }
 
+# On x86_64, we retrieve the hardware info from SMBIOS, unless it has been
+# provided via the environment
+if test -d /sys/class/dmi/id; then
+        test -z "$HWD_VENDOR" && HWD_VENDOR=$(cat /sys/class/dmi/id/sys_vendor)
```

Comment:
I think this script is not (yet) tainted by `set -e`, but generally a safer pattern is to `test faultystuff || recoverystuff` (e.g. `test -n "$value" || value="..."` so the line ends up successfully either way - or there is a bigger failure in play.